packages/@ember/object/index.ts

```diff
@@ -130,18 +131,43 @@ export {
 
 const BINDINGS_MAP = new WeakMap();
 
-function setupAction(target, key, actionFn) {
-  if (target.constructor !== undefined && typeof target.constructor.proto === 'function') {
+interface HasProto {
+  constructor: {
+    proto(): void;
+  };
+}
+
+function hasProto(obj: unknown): obj is HasProto {
+  return (
+    obj != null &&
+    (obj as any).constructor !== undefined &&
+    typeof ((obj as any).constructor as any).proto === 'function'
+  );
+}
+
+interface HasActions {
+  actions: Record<string | symbol, unknown>;
+}
+
+function setupAction(
+  target: object,
+  key: string | symbol,
+  actionFn: Function
+): TypedPropertyDescriptor<unknown> {
+  // SAFETY: Accessing an unknown property on a non-null value isn't unsafe.
```

Comment:
I think we don't need the comment now that it has the guard!

Reply:
Correct!